src/main/java/edu/harvard/iq/dataverse/FileDownloadServiceBean.java

```diff
@@ -235,6 +235,10 @@ public void explore(GuestbookResponse guestbookResponse, FileMetadata fmd, Exter
                 dataFile = guestbookResponse.getDataFile();
             }
         }
+        //For tools to get the dataset and datasetversion ids, we need a full DataFile object (not a findCheapAndEasy() copy)
```

Comment:
@qqmyers this looks fine but have you seen any performance issue here? If so, we could right some helper method like doesExternalToolNeedDataset(externalTool). But if there's trivial performance impact, not worth it.

Reply:
@scolapasta - I think it's trivial - this is only called during tool launch and it is then a lookup of one datafile object. I didn't see any obvious delay launching a tool and I think the main concerns previously have been for cases where many datafiles were being looked up, e.g. when scanning through the list in a dataset...

Reply:
OK< great. That is what I assumed as well, and why I had already moved to QA. :)